signer/src/stacks/api.rs

```diff
@@ -902,11 +896,23 @@ impl StacksInteract for StacksClient {
     where
         T: AsTxPayload + Send + Sync,
     {
+        // Consensus serialize the transaction payload to bytes. This is the
+        // method that the stacks node uses to determine transaction size when
+        // verifying admittance into the mempool.
+        let mut payload_bytes = Vec::new();
+        payload
+            .tx_payload()
+            .consensus_serialize(&mut payload_bytes)
+            .map_err(Error::StacksCodec)?;
+        // In Stacks core, the minimum fee is 1 mSTX per byte, so we double
+        // that here as a precaution and cap it at our maximum fee.
+        let default_min_fee = (payload_bytes.len() as u64 * 2).min(MAX_TX_FEE);
```

Comment:
I wasn't quite sure if we need another multiplier here, you had 1_000_000 as the default fee and this will definitely not be 1 million. The stacks code only has "1" listed, and doesn't say if that's mstx or stx.